### DISCUSSION TITLE
Uninstall FileZilla

### DISCUSSION THREAD

**Discussion by Romanitho** [Romanitho/Winget-Install#29] [General]
@KnifMelti I have a question regarding the way you uninstall FileZilla. You run a Mod script to wait for the uninstall popup to be clicked on, right ? Cause in a system context, this popup is not visible... 🤔

**Comment by KnifMelti** [comment]
>Well, it is (in production too)...
![Winget-Install via PS](https://user-images.githubusercontent.com/102996177/164889182-f0334161-88a2-49b9-af78-a7a2869e5774.png)
![Winget-Install via CMD](https://user-images.githubusercontent.com/102996177/164889196-abdaac05-6640-4b9a-8071-629ae45a80d7.png)

**Reply by Romanitho** [reply]
because you use a tool that display system processes, no ? but with Intune or SCCM, you will probably not see this window.

**Reply by KnifMelti** [reply]
My GetKing.cmd is just:
powershell.exe -ExecutionPolicy bypass -File "%~dp0winget-install.ps1" %*

It makes it cleaner in SCCM and we just point and click when creating ion our framework

**Reply by KnifMelti** [reply]
No, it's like this in SCCM too - we run it in production (Intune I don't know)

**Reply by Romanitho** [reply]
I made a test on my machine. If i run (as a scheduled task as system)
`"C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.18.431.0_x64__8wekyb3d8bbwe\winget.exe" uninstall TimKosse.FileZilla.Client`
I don't see any unisntall windows in my user session, but in my processes, I have the uninstaller waiting for a "humain" action :
![image](https://user-images.githubusercontent.com/96626929/164889924-e0aefcdf-15c4-46d0-813e-4500c594d3cd.png)
I have to kill it

**Reply by KnifMelti** [reply]
Yes, when your calling winget directly, but tha's not the case with winget-install -Uninstall.
Its PS that runs a PS-file

**Reply by Romanitho** [reply]
sure, but everything is under system and should remain in system.

**Reply by KnifMelti** [reply]
Well, it is like this in our "swedish" SCCM ;)
We have the Winget installations is Software Center and our users can install/uninstall them when they like

**Reply by Romanitho** [reply]
that's why i didn't recommand to use my script with uninstall, as Winget doesn't always know how to handle uninstallers.

**Reply by Romanitho** [reply]
> Well, it is like this in our "swedish" SCCM ;) We have the Winget installations is Software Center and our users can install/uninstall them when they like

I cannot really understand how your users can natively have uninstall window in system context

**Reply by KnifMelti** [reply]
Maybe it's a flaw in SCCM (cause I cant reproduce it via task scheduler here in system context - it's hidden from the user in system context)?

**Reply by KnifMelti** [reply]
In SCCM when you presents it in Software Center as available it gives the users a button to instell/uninstall whenever they like. But it runs in system context via SCCM agent

**Reply by KnifMelti** [reply]
Maybe give another example in the ReadMe?

**Reply by Romanitho** [reply]
In SCCM, in system context, no window is shown. Except if I use ServiceUI.exe in propose, that can show the system process to a user session ID (That I use to troubleshoot for instance). But by default, in system context you don't have access as user. Fortunately :)

**Reply by Romanitho** [reply]
> In SCCM when you presents it in Software Center as available it gives the users a button to instell/uninstall whenever they like. But it runs in system context via SCCM agent

100% agree. But no process is shown to user. So for example, if a user want to uninstall FileZilla because it was available, it will not work with our winget-install script as it show the uninstaller window.

**Reply by Romanitho** [reply]
> Maybe give another example in the ReadMe?

Maybe we can pick an example for installation, yes.
The thing is, we can keep your mods feature anyway as it can be interesting obviously. For uninstalls, I recommend to use the original ones for system context. But i know that on winget side, they work on better handling uninstallations.

**Reply by KnifMelti** [reply]
Well, I can manufacture an installation that deletes the Desktop icon for the program (that we don't want)

**Comment by KnifMelti** [comment]
Heres the reality (with our old solution built on your scripts):

![SCClient_k41CmB6CVG](https://user-images.githubusercontent.com/102996177/164890941-811b1457-2b68-4868-b779-f5aff196bd22.png)

![mRemoteNG_7rDffNTKxW](https://user-images.githubusercontent.com/102996177/164891019-ecd7b33a-5c16-48fe-bae9-3d0d20f6ca0b.png)

![mRemoteNG_V5SRDst6Kz](https://user-images.githubusercontent.com/102996177/164891058-83624fc0-aaa8-493c-8d4d-a8cd0140d6c1.png)

**Reply by Romanitho** [reply]
So weird 😄

**Comment by KnifMelti** [comment]
Wev'e been using this available way in deployments (not a mandatory **deployment**) since forever.
You maybe only works with mandatory **deployments**?

**Reply by Romanitho** [reply]
no, both of them. Required and Available. But in system context, nothing is shown to user. You might use another tool or something

**Reply by KnifMelti** [reply]
no no, its just vanilla SCCM (I'm one of the responsible for the system)

**Reply by Romanitho** [reply]
and I am an integrator. Work with multiple customers and different SCCM infrastructures :) but it's the first time I see that behaviour, that's why I don't really understand

**Reply by KnifMelti** [reply]
And all our surrounding IT-apartments use it too :)

**Reply by Romanitho** [reply]
your GetKing.cmd only run our script ? really strange haha. Really interested in your case :)

**Reply by KnifMelti** [reply]
I can give many other examples as available in system context presenting a graphical installation (we always have to make sure they doesn't give the users a file selector, links etc because as system...)

**Reply by KnifMelti** [reply]
And that's ordinary install, not this solution

**Reply by Romanitho** [reply]
oh no, shame on me. I always use full silent install/uninstall...

**Reply by Romanitho** [reply]
and under Intune, it doesn't show any interactive window under system. I'm too tired 😅

**Reply by KnifMelti** [reply]
I'll compile some applications we use this technique on when I'm back at work - it's really useful!

**Reply by Romanitho** [reply]
![image](https://user-images.githubusercontent.com/96626929/164913013-e623cc90-c918-454e-87a1-ee87a2a5e60a.png)
When I remove my silent switch, I have the notepad++ uninstall prompt as well, even as system. I didn't remember that under SCCM, my bad ;)  
but I always use quiet or silent switches to avoid any prompt. A reflex ^^
It means that by default SCCM uses ServiceUI.exe or something, not native behaviour under Intune (or I say shi* again 😅)

by the way, that's why I use winget-install script to install, but not to uninstall https://github.com/Romanitho/Winget-Install#uninstall
because in your context, you are waiting for actions from end user. Not really like this for our users

**Reply by KnifMelti** [reply]
Glad we agree on the SCCM **available deploymen**t function! :=)
Well, it's up to our users when to install/uninstall the whitelisted WinGet applications from Software Center.
And via winget uninstall you get this behavior from some applications, that's why I came up with the Mods solution in the first place (that and the fact I wanted a way to delete desktop shortcuts/populate the registry/copy som settings file etc) in this **framework** we're using...
...and this takes care of the actions from end users when it comes to FileZilla uninstall :)

**Reply by KnifMelti** [reply]
...I've also seen that winget is going to get better support for uninstallation of applications in the future.
That's good!

**Reply by Romanitho** [reply]
that's why as a workaround  suggested a homemade uninstall way. either the native uninstall (like `"C:\Program Files\Notepad++\uninstall.exe" /S` for Notepad++) or custom script (if you want to do more :) ).
At the end, you added a feature that can help for more than just uninstall shity apps ^^

**Reply by KnifMelti** [reply]
And, your idea of an install-once was excellent!
